pandas/core/generic.py

```diff
@@ -1940,6 +1940,79 @@ def tail(self, n=5):
             return self
         return self.iloc[-n:]
 
+  
+    def sample(self, n=None, frac=None , replace=False, weights=None, seed=None):
+        """
+        Returns a sample of rows from object. 
+        
+        Parameters
+        ----------
+        n: in, optional, Default = 5 if frac = None. 
+            Number of rows to return. Cannot be used with frac.
+        frac: float. Default = None, S
+            Share of rows to return. Cannot be used with n. 
+        replace: boolean. Default = False. 
+            Sample with or without replacement. 
+        weights: Series, list, or ndarray. Default = None.
+            Default 'None' results in equal probability weighting. 
+            If called on a DataFrame or Panel, will also accept the name of a 
+            column as a string. Must be same length as index. 
+            Weights must be normalized. 
+        seed: int. Default = None. 
+            seed to be fed to numpy random.RandomState() Function.
+            
+        Returns
+        -------
+        Frame of same type as caller. 
+
+        Notes
+        -----
+        When called without arguments (.sample()), returns 5 random columns
+        in a manner analogous to .head() or .tail(). 
+
+        -----------
+        """
+    
+        # Setup vars
+        rs = np.random.RandomState(seed)
+    
+        # Check weights for compliance
+        if weights is not None:
+                                      
+            # Strings acceptable if not a series
+            if isinstance(weights, string_types) and self.ndim >1 :
+                try:
+                    weights = self[weights]
+                except(KeyError):
+                    raise KeyError("String passed to weights not a valid column name")
+            
+            if isinstance(weights, string_types) and self.ndim == 1:
+                raise ValueError("Strings cannot be passed as weights when sampling from a Series.")
+
+            #normalize format of weights to ndarray. 
+            weights = np.asarray(weights)
+                
+            # Check length (numpy does this, but has confusing errors with different argument labels.)
+            if len(weights) != len(self):
+                raise ValueError("Weights and data to be sampled must be of same length")
+
+    
+        # Check whether frac or N
+        if n is None and frac is None:
+            n = 5
+        elif n is None and frac is not None:
+            n = int(round(frac * len(self))) 
+        elif n is not None and frac is not None:
+            raise ValueError('Please enter a value for "frac" OR "n", not both')
```

Comment:
I think variables are typically put in single backticks. [Here's](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#method-docstrings) all the rules.